projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -173,19 +211,26 @@ public function get_site_sub_resource_from_wpcom( $req ) {
 	 * @return array
 	 */
 	public function get_stats_single_resource_from_wpcom( $req ) {
-		return static::request_as_blog_cached(
-			sprintf(
-				'/sites/%d/stats/%s/%d?%s',
-				Jetpack_Options::get_option( 'id' ),
-				$req->get_param( 'resource' ),
-				$req->get_param( 'resource_id' ),
-				http_build_query(
-					$req->get_params()
-				)
-			),
-			'1.1',
-			array( 'timeout' => 30 )
-		);
+		switch ( $req->get_param( 'resource' ) ) {
+			case 'post':
+				return $this->wpcom_stats->get_post_views(
+					intval( $req->get_param( 'resource_id' ) ),
+					http_build_query(
+						$req->get_params()
+					)
+				);
+
+			case 'video':
+				return $this->wpcom_stats->get_video_details(
```

Comment:
It seems the naming is not very consistent.

Reply:
I guess you mean `get_video_details`, compared to `get_post_views`?

Reply:
This is another conversation, closing it here.